# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/meson/revert-cmdline-limit.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- meson-1.6.0/mesonbuild/backend/ninjabackend.py	2024-10-20 20:20:25.000000000 +0200
+++ meson-1.5.0/mesonbuild/backend/ninjabackend.py	2024-07-10 19:25:01.000000000 +0200
@@ -101,11 +98,8 @@
         # and that has a limit of 8k.
         limit = 8192
     else:
-        # Unix-like OSes usually have very large command line limits, (On Linux,
-        # for example, this is limited by the kernel's MAX_ARG_STRLEN). However,
-        # some programs place much lower limits, notably Wine which enforces a
-        # 32k limit like Windows. Therefore, we limit the command line to 32k.
-        limit = 32768
-    # Be conservative
-    limit = limit // 2
+        # On Linux, ninja always passes the commandline as a single
+        # big string to /bin/sh, and the kernel limits the size of a
+        # single argument; see MAX_ARG_STRLEN
+        limit = 131072
     return int(os.environ.get('MESON_RSP_THRESHOLD', limit))
